app/src/main/java/com/hwang/xsighting/CreateSightingFragment.java

```diff
@@ -0,0 +1,446 @@
+  private FirebaseFirestore db = FirebaseFirestore.getInstance();
+  private FusedLocationProviderClient fusedLocationClient;
+  private final int MY_PERMISSIONS_REQUEST_LOCATIONS = 1122;
+  private String lastLocation;
+  private GeoPoint geoPointLocation;
+  private FirebaseUser user;
+  private Context mContext;
+  private Activity mActivity;
+
+  // Picture variables
+  // https://github.com/NJCrain/health-tracker/blob/master/app/src/main/java/com/njcrain/android/healthtracker/activity/ProfileActivity.java
+  // https://developer.android.com/training/camera/photobasics
+  // https://developer.android.com/training/permissions/requesting
+  private final int MY_PERMISSIONS_ID = 0;
+  private boolean CAMERA_PERMISSION_GRANTED = false;
+  private boolean FILES_PERMISSION_GRANTED = false;
+  static final int PICK_IMAGE = 2;
+  private int REQUEST_TAKE_PHOTO = 3;
+  String currentPhotoPath = null;
+  ImageView sightingImage;
+
+
+
+
+  public CreateSightingFragment() {
+    // Required empty public constructor
+  }
+
+  public static CreateSightingFragment newInstance(String param1, String param2){
+    CreateSightingFragment fragment = new CreateSightingFragment();
+    Bundle args = new Bundle();
+    fragment.setArguments(args);
+    return fragment;
+  }
+
+  @Override
+  public void onCreate(Bundle savedInstanceState){
+    super.onCreate(savedInstanceState);
+  }
+
+  @Override
+  public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState){
+    View view = inflater.inflate(R.layout.fragment_create_sighting, container, false);
+    mActivity = getActivity();
+    mContext = view.getContext();
+
+    // Gets the imageView that will be updated when the user uploads a picture
+    sightingImage = view.findViewById(R.id.sightingImage);
+
+    // Get Location
+    fusedLocationClient = LocationServices.getFusedLocationProviderClient(view.getContext());
+    Log.i("LOCATION", "CALLING GET LOCATION");
+    getLocation();
+
+    // Set the username
+    user = FirebaseAuth.getInstance().getCurrentUser();
+
+    view.findViewById(R.id.button_submit_report).setOnClickListener(mListener);
+    view.findViewById(R.id.takePictureButton).setOnClickListener(mListener);
+    view.findViewById(R.id.choosePictureButton).setOnClickListener(mListener);
+
+
+    return view;
+  }
+
+  private final View.OnClickListener mListener = new View.OnClickListener(){
+    public void onClick(View view){
+      switch (view.getId()){
+        case R.id.button_submit_report:
+          submitSighting();
+          break;
+        case R.id.takePictureButton:
+          takePicture();
+          break;
+        case R.id.choosePictureButton:
+          choosePictureFromFiles();
+          break;
+      }
+    }
+  };
+
+  //TODO: throws exception on super, Commenting out because I am uncertain on what this does and it doesn't seem to break anything
+//  @Override
+//  protected void onRestart(){

```

Comment:
I'm not exactly sure what this method does. 

All I know is that it is throwing an error and when commented out functionality seems to stay the same. Please advise